include/zephyr/zephyr.h

```diff
@@ -9,8 +9,4 @@
 
 #include <zephyr/kernel.h>
 
-#ifndef CONFIG_LEGACY_INCLUDE_PATH
-#warning "<zephyr/zephyr.h> is deprecated, include <zephyr/kernel.h> instead"
-#endif
-
```

Comment:
#warning should be kept

Reply:
If the Kconfig is removed, the project should fail to configure if the option is present, so it wouldn't get to the stage of compiling this

Reply:
We need to allow usage of `<zephyr/zephyr.h>` until deprecation period expires (until post 3.4 in this case). It was decided not to warn users if they were using legacy includes. So as this option is removed now, users need to get the warning unconditionally.

Reply:
I forgot about that, indeed!

Reply:
Sorry I was just dropping every block used by that config, did not realize that we reused it for the zephyr.h warning. Thanks for the pointer.